x/merkledb/history.go

```diff
@@ -249,6 +254,13 @@ func (th *trieHistory) getChangesToGetToRoot(rootID ids.ID, start maybe.Maybe[[]
 	for i := mostRecentChangeIndex; i > lastRootChangeIndex; i-- {
 		changes, _ := th.history.Index(i)
 
+		if i == mostRecentChangeIndex {
```

Comment:
should these just be outside the loop?

Reply:
They could be, but then we'd end up having to get it from `th.history` again anyway, so I don't think that'd really buy us anything